ocaml-lsp-server/src/inlay_hints.ml

```diff
@@ -0,0 +1,108 @@
+open Import
+open Fiber.O
+
+let overlaps (x : Range.t) (y : Range.t) =
+  let open Ordering in
+  match (Position.compare x.start y.end_, Position.compare x.end_ y.start) with
+  | (Lt | Eq), (Gt | Eq) | (Gt | Eq), (Lt | Eq) -> true
+  | _ -> false
+
+let range_overlaps_loc range loc =
+  match Range.of_loc_opt loc with
+  | Some range' -> overlaps range range'
+  | None -> false
+
+let outline_type ~env typ =
+  let ppf, to_string = Format.to_string () in
+  Ocaml_typing.Printtyp.wrap_printing_env env (fun () ->
+      Merlin_analysis.Type_utils.print_type_with_decl
+        ~verbosity:(Mconfig.Verbosity.Lvl 0)
+        env
+        ppf
+        typ);
+  Some (sprintf ": %s" (to_string ()))
```

Comment:
The printing might need some processing, I have weird symbol appearing in my editor:
<img width="660" alt="image" src="https://github.com/ocaml/ocaml-lsp/assets/5031221/a5ced989-1004-42a1-b552-412b8f8dc3ab">


I guess these are linebreaks that should be replaced by a space (and maybe mean the hint is too long anyway...)

Reply:
I think I lifted `outline_type` from somewhere else. Looks like its behavior isn't quite right for this use.